core/sql/optimizer/BindItemExpr.cpp

```diff
@@ -6105,12 +6105,21 @@ ItemExpr *Assign::bindNode(BindWA *bindWA)
               ItemExpr *newChild;
               const NAType &desiredType = child(0)->getValueId().getType();
               SQLBlob &lobType = (SQLBlob&)desiredType;
-
-              NAType * newType = new SQLBlob((CmpCommon::getDefaultNumeric(LOB_MAX_SIZE)*1024*1024), 
+              short fs_datatype = child(0)->castToItemExpr()->getValueId().getType().getFSDatatype();
+
+              NAType * newType = NULL;
+              if (fs_datatype == REC_CLOB) {
+                  newType = new SQLClob((CmpCommon::getDefaultNumeric(LOB_MAX_SIZE) * 1024 * 1024),
+                         lobType.getLobStorage(),
+                         TRUE, FALSE, TRUE,
+                         CmpCommon::getDefaultNumeric(LOB_BATCH_SIZE));
+              }
+              else {
+              newType = new SQLBlob((CmpCommon::getDefaultNumeric(LOB_MAX_SIZE)*1024*1024),
                                              lobType.getLobStorage(), 
                                              TRUE, FALSE, TRUE, 
-                                             CmpCommon::getDefaultNumeric(LOB_MAX_CHUNK_MEM_SIZE)*1024*1024); 
-              //              vid1.coerceType(desiredType, NA_LOB_TYPE); 
+                                             CmpCommon::getDefaultNumeric(LOB_BATCH_SIZE));
+              }
```

Comment:
I think it's better to compare the CQD and this 'param'. If the param is larger, I will use CQD, else use the param.  I will change it.